trainers/base_trainer.py

```diff
@@ -43,18 +79,19 @@ def __init__(
         self.cfg = cfg
         self.gradient_accumulation_steps = cfg["trainer"]["training"][
             "gradient_accumulation_steps"
-        ]
+        ] // torch.cuda.device_count() ## divide by number of GPUs to maximise throughput
```

Comment:
what happens when we have say an odd number of grad accumulation steps and even number of gpus (or more likely 3 gpus and say 20 gpu acc. steps)

Reply:
tbh not sure what the best idea here is ... it may be that we want to assert that it is divisible by 3 or sth. not sure